src/embezzler.ts

```diff
@@ -169,6 +180,127 @@ export const embezzlerSources = [
     [],
     true
   ),
+  new EmbezzlerFight(
+    "Time-Spinner",
+    () =>
+      have($item`Time-Spinner`) &&
+      cliExecuteOutput("timespinner list monsters").includes(
+        $monster`Knob Goblin Embezzler`.name
+      ) &&
+      get("_timeSpinnerMinutesUsed") <= 7,
+    () =>
+      have($item`Time-Spinner`) &&
+      cliExecuteOutput("timespinner list monsters").includes($monster`Knob Goblin Embezzler`.name)
+        ? Math.min((10 - get("_timeSpinnerMinutesUsed")) / 3)
+        : 0,
+    (options: EmbezzlerFightOptions) => {
+      const macro = options.macro ?? embezzlerMacro();
+      withMacro(macro, () => {
+        visitUrl(`inv_use.php?whichitem=${toInt($item`Time-Spinner`)}`);
+        runChoice(1);
+        visitUrl(
+          `choice.php?whichchoice=1196&monid=${$monster`Knob Goblin Embezzler`.id}&option=1`
+        );
+      });
+    }
+  ),
+  new EmbezzlerFight(
+    "Macrometeorite",
+    () =>
+      get("beGregariousMonster") === $monster`Knob Goblin Embezzler` &&
+      get("beGregariousFightsLeft") > 0 &&
+      have($skill`Meteor Lore`) &&
+      get("_macrometeoriteUses") < 10,
+    () =>
+      get("beGregariousMonster") === $monster`Knob Goblin Embezzler` &&
+      get("beGregariousFightsLeft") > 0 &&
+      have($skill`Meteor Lore`)
+        ? 10 - get("_macrometeoriteUses")
+        : 0,
+    (options: EmbezzlerFightOptions) => {
+      saberCrateIfDesired();
+      if (
+        have($item`miniature crystal ball`) &&
+        get("crystalBallLocation") === $location`Noob Cave` &&
+        get("crystalBallMonster") !== $monster`Knob Goblin Embezzler` &&
+        !(get("_saberForceMonster") === $monster`crate` && get("_saberForceMonsterCount") > 0)
+      ) {
+        equip($slot`familiar`, $item`miniature crystal ball`);
+      }
+      const baseMacro = options.macro ?? embezzlerMacro();
+      const macro = Macro.if_($monster`crate`, Macro.skill($skill`Macrometeorite`)).step(baseMacro);
+      adventureMacro($location`Noob Cave`, macro);
+    }
+    //do we want to equip orb on these guys?
```

Comment:
Theoretically, we can get an extra one by putting the orb on for the final greg fight. Whether or not that is EV positive depends on how many macros you have left when you have one greg left and how valueable your familiar equipment slot is. 
Might be worth to rule of thumb yes and then use other zones like barf to wipe the prediction if it is ever an embezzler as opposed to a crate. Until the end of our proposed chain when we do want one last Embez prediction

Reply:
I think we use orb on the final warren fight, not noob cave; are you saying we'd get another one?

I think I don't want to start interpolating barf turns, but I can add some sort of check to the available() function for the two replacement guys, that way we can do other embezzler fights to reset orb. The only thing is we need to find a zone other than noob cave for our wanderers to default to if we're going to do that--backups could be dangerous.

Reply:
You are correct. I misunderstood the zone change. You can't get another extra because the Embez isn't a free fight.

I'm not sure I fully understand what you mean in the second section. But yes. The idea is that when we predict an Embez and we are not done doing macros, we don't want to hit the greg'd Embez. So we want to do some turn costing action to wipe the prediction. We would really like it to be another Embez bc we are already mega buffed. 
Is the problem with fighting wanderers in the noobcave that it might confuse us about which source we got the Embez from?

Reply:
I'm not sure if there's a specific thing about going to noob cave outside of this context that puts me on edge, but not defaulting to it would generally be safer for probably a lot of reasons.